core/src/main/java/org/bitcoinj/core/LockTime.java

```diff
@@ -155,8 +187,14 @@ public int hashCode() {
 
     @Override
     public String toString() {
-        return isBlockHeight() ?
-                "block " + blockHeight() :
-                TimeUtils.dateTimeFormat(timestamp());
+        String result;
+        if (this instanceof NoLock) {
+            result = "no lock";
+        } else if (this instanceof HeightLock) {
+            result =  "block " + ((HeightLock)this).blockHeight();
+        } else {
+            result = TimeUtils.dateTimeFormat(((TimeLock)this).timestamp());
+        }
+        return result;
```

Comment:
Sadly we can't use pattern matching here.

I would use 3 return statements and get rid of 2 lines…

Reply:
Fair enough. We should probably have a conversation about the pros and cons of the various idioms we have available pre-pattern-matching and try to standardize a little. I deliberately used several of the alternatives in the PR just for comparison.